monkestation/code/modules/smithing/anvil/anvil.dm

```diff
@@ -37,47 +37,68 @@
 
 /obj/structure/anvil/attack_hand(mob/living/user, list/modifiers)
 	. = ..()
-	if(!chosen_recipe)
-		var/pick = show_radial_menu(user, src, recipes, custom_check = FALSE, require_near = TRUE, tooltips = TRUE)
-		if(!pick)
-			return
-		if(!(pick in name_to_type))
-			return
-		chosen_recipe = name_to_type[pick]
+	if(anchored && !smithing)
+		if(!chosen_recipe)
+			var/pick = show_radial_menu(user, src, recipes, custom_check = FALSE, require_near = TRUE, tooltips = TRUE)
+			if(!pick)
+				return
+			if(!(pick in name_to_type))
+				return
+			chosen_recipe = name_to_type[pick]
 
-	if(!smithing && working_material && chosen_recipe && working_material.material_stats)
-		var/density_hardness = 0
-		density_hardness = working_material.material_stats.hardness + working_material.material_stats.density
+		if(working_material && chosen_recipe && working_material.material_stats)
+			var/density_hardness = 0
+			density_hardness = working_material.material_stats.hardness + working_material.material_stats.density
 
-		var/difficulty_modifier = density_hardness / 30
+			var/difficulty_modifier = density_hardness / 30
 
-		new /datum/anvil_challenge(src, new chosen_recipe, user, difficulty_modifier)
-		smithing = TRUE
+			new /datum/anvil_challenge(src, new chosen_recipe, user, difficulty_modifier)
+			smithing = TRUE
 
 /obj/structure/anvil/attack_hand_secondary(mob/user, list/modifiers)
-	if(chosen_recipe)
+	if(chosen_recipe && !smithing)
 		chosen_recipe = null
 		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 	. = ..()
 
 /obj/structure/anvil/proc/generate_item(quality)
-	var/obj/item/smithed_part/new_part = chosen_recipe.output
-	new new_part (get_turf(src), working_material, quality)
-	QDEL_NULL(working_material)
+	var/obj/output = new chosen_recipe.output
+	if(istype(output,/obj/item/smithed_part))
+		var/obj/item/smithed_part/new_part = chosen_recipe.output
+		new new_part (get_turf(src), working_material, quality)
+		QDEL_NULL(working_material)
+	else if(istype(output,/obj/item/clothing/smithed_clothes))
+		var/obj/item/clothing/smithed_clothes/new_part = chosen_recipe.output
+		new new_part (get_turf(src), working_material, quality)
+		QDEL_NULL(working_material)
+	QDEL_NULL(output) //Thanks, bye
 
 /obj/structure/anvil/attackby(obj/item/attacking_item, mob/living/user, params)
 	if((isstack(attacking_item) || istype(attacking_item, /obj/item/merged_material)) && !smithing)
 		if(try_place_item(attacking_item, user))
 			return
+	if(attacking_item.tool_behaviour == TOOL_WRENCH && !smithing)
```

Comment:
should be able to just use default wrench act for these

Reply:
Did thing